src/web/web-service/wpan_service.hpp

```diff
@@ -0,0 +1,169 @@
+#include "common/logging.hpp"
+#include "utils/hex.hpp"
+
+/**
+ * WPAN parameter constants
+ *
+ */
+#define OT_BORDER_ROUTER_PORT 49191
+#define OT_EXTENDED_PANID_LENGTH 8
+#define OT_HARDWARE_ADDRESS_LENGTH 8
+#define OT_NETWORK_NAME_LENGTH 16
+#define OT_PANID_LENGTH 4
+#define OT_PSKC_MAX_LENGTH 16
+#define OT_PUBLISH_SERVICE_INTERVAL 20
+
+namespace ot {
+namespace Web {
+
+/**
+ * This class provides web service to manage WPAN.
+ *
+ */
+class WpanService
+{
+public:
+
+    /**
+     * This method handles the http request to join network.
+     *
+     * @returns The reference to the http response of joining network.
+     *
+     */
+    std::string &HandleJoinNetworkRequest(const std::string &aJoinRequest);
+
+    /**
+     * This method handles the http request to form network.
+     *
+     * @returns The reference to the http response of forming network.
+     *
+     */
+    std::string &HandleFormNetworkRequest(const std::string &aFormRequest);
+
+    /**
+     * This method handles the http request to add on-mesh prefix.
+     *
+     * @returns The reference to the http response of adding on-mesh prefix.
+     *
+     */
+    std::string &HandleAddPrefixRequest(const std::string &aAddPrefixRequest);
+
+    /**
+     * This method handles the http request to delete on-mesh prefix http request.
+     *
+     * @returns The reference to the http response of deleting on-mesh prefix.
+     *
+     */
+    std::string &HandleDeletePrefixRequest(const std::string &aDeleteRequest);
+
+    /**
+     * This method handles http request to get netowrk status.
+     *
+     * @returns The reference to the http response of getting status.
+     *
+     */
+    std::string &HandleStatusRequest(void);
+
+    /**
+     * This method handles http request to get available networks.
+     *
+     * @returns The reference to the http response of getting available networks.
+     *
+     */
+    std::string &HandleAvailableNetworkRequest(void);
+
+    /**
+     * This method sets the interface name of the wpantund.
+     *
+     * @param[in]  aIfName  The pointer to the interface name of wpantund.
+     *
+     */
+    void SetInterfaceName(const char *aIfName) { strncpy(mIfName, aIfName, sizeof(mIfName)); }
+
+    /**
+     * This method gets status of wpan service.
+     *
+     * @param[in]  aNetworkName  The pointer to the network name.
+     * @param[in]  aIfName  The pointer to the extended PAN ID.
+     *
+     * @retval kWpanStatus_OK        Started the wpan service.
+     * @retval kWpanStatus_OffLine   Not started the wpan service.
+     * @retval kWpanStatus_Down      The wpantund was down.
+     *
+     */
+    int GetWpanServiceStatus(std::string &aNetworkName, std::string &aExtPanId) const;
+
+private:
+
+    ot::Dbus::WpanNetworkInfo mNetworks[DBUS_MAXIMUM_NAME_LENGTH];
+    int                       mNetworksCount;
+    char                      mIfName[IFNAMSIZ];
+    std::string               mNetworkName, mExtPanId;
```

Comment:
Although not listed in the code style, it seems OpenThread defines member variables one by one.